### ISSUE TITLE
gridSaveState 

### ISSUE THREAD

**Issue by vdiazrj** [angular-ui/ui-grid#5250]
If there are filters applied to columns the result of gridApi.saveState.save() strips the condition out of the objects in the column.filters array. When attempting to reapply the filters, it doesn't work because of this missing property.

To reproduce:
1. Generate a grid
2. Apply a filter to a column (e.g. gridApi.grid.columns[2].filters = [{term: 'foo', condition: 8}]
3. Execute gridApi.grid.refresh();
4. Save the results of saveState.save() into var (e.g. var bar = gridApi.saveState.save())
5. Examine bar.columns[2].filters === [{term: 'foo'}] i.e condition property is missing.

In Plnkr http://plnkr.co/edit/jHHMdZSez3Vxui4VWbP4?p=preview:
1. Enter search term in input field
2. Examine $scope.gridApi.grid.columns[4].filters after $scope.search has been executed
3. Examine the filter for the colummn in $scope.state.column after $scope.saveState() has been executed
4. Note that condition property is missing from the filter applied to the column in $scope.state

**Comment by tbaeg** [comment]
Reference this commit.

https://github.com/angular-ui/ui-grid/commit/eabff1e18a621839b3020972452d918e9b88b181

Looks like it was 'fixed' that way but not entirely sure why. Will need to look at the source code to confirm.

**Comment by dKab** [comment]
I totally agree that stripping `condition` from filters is very confusing behavior. When I copy filters, I expect them to be copied fully. This was implemented the way it works now in order to address some edge case.

**Comment by orneryd** [comment]
Thanks for contributing! This repository has been transferred, and we're preparing the new 5.x release. We're closing older issues and pull requests as part of the reset. Please join us on Discord to follow the work or continue the conversation: https://discord.gg/Baz4w8ZWN
